src/crimson/admin/osd_admin.cc

```diff
@@ -93,6 +93,89 @@ class SendBeaconHook : public AdminSocketHook {
 template std::unique_ptr<AdminSocketHook>
 make_asok_hook<SendBeaconHook>(crimson::osd::OSD& osd);
 
+/**
+ * An OSD admin hook: run bench
+ * Usage parameters:
+ *   count=Count of bytes to write
+ *   bsize=block size
+ *   osize=Object size
+ *   onum=Number of objects
+ */
+class RunOSDBenchHook : public AdminSocketHook {
+public:
+  explicit RunOSDBenchHook(crimson::osd::OSD& osd) :
+    AdminSocketHook{"bench",
+        "name=count,type=CephInt,req=false "
+        "name=size,type=CephInt,req=false "
+        "name=object_size,type=CephInt,req=false "
+        "name=object_num,type=CephInt,req=false",
+        "run OSD bench"},
+    osd(osd)
+  {}
+  seastar::future<tell_result_t> call(const cmdmap_t& cmdmap,
+              std::string_view format,
+              ceph::bufferlist&& input) const final
+  {
+    int64_t count = cmd_getval_or<int64_t>(cmdmap, "count", 1LL << 30);
+    int64_t bsize = cmd_getval_or<int64_t>(cmdmap, "size", 4LL << 20);
+    int64_t osize = cmd_getval_or<int64_t>(cmdmap, "object_size", 0);
+    int64_t onum = cmd_getval_or<int64_t>(cmdmap, "object_num", 0);
+    double elapsed = 0.0;
+    auto duration = local_conf()->osd_bench_duration;
+
+    if (bsize > local_conf()->osd_bench_max_block_size) {
+      // let us limit the block size because the next checks rely on it
+      // having a sane value.  If we allow any block size to be set things
+      // can still go sideways.
+      logger().info("block 'size' values are capped at {}. If you wish to use"
+                    " a higher value, please adjust 'osd_bench_max_block_size'",
+                    byte_u_t(local_conf()->osd_bench_max_block_size));
+      return seastar::make_ready_future<tell_result_t>(-EINVAL, "block size too large");
+    } else if (bsize < (1LL << 20)) {
+      int64_t max_count = bsize * duration * local_conf()->osd_bench_small_size_max_iops;
+      if (count > max_count) {
+        logger().info("bench count {} > osd_bench_small_size_max_iops {}",
```

Comment:
See `crimson/common/log.h`, can you please use `INFO` and `DEBUG` macros?

Reply:
i don't mind to use INFO or DEBUG, but all the other using logger() in osd_admin , will change that

Reply:
These macros were introduced recently, we are moving gradually :)